### ISSUE TITLE
NullPointerException: Registry entry not present

### ISSUE THREAD

**Issue by DadoIrie** [NovaStarter/Create-Crystal-Clear#26]
Now this crash is kinda hard to pinpoint as here (in this linked crashlog) regardless which mod I'm removing that crash doesn't happen, but if all of em are loaded the crash happens. 
L253 `Caught exception during event RegisterEvent dispatch for modid create_crystal_clear java.lang.NullPointerException: Registry entry not present: create:cogwheel` just before L195 `Exception caught during firing event: Registry entry not present: create:cogwheel`

1st crashlog (as explained above)
https://mclo.gs/fyOynr0

Well the, I decided I can live with the experimental features popping up when launching world and will just remove the yeetus mod from my modpack and this happens:
https://mclo.gs/xJQiDGZ

https://mclo.gs/xJQiDGZ#L1336 (`Exception caught during firing event: Registry entry not present: create:cogwheel`) 1st related error which leads to crash
https://mclo.gs/xJQiDGZ#L1394 (`Caught exception during event RegisterEvent dispatch for modid create_crystal_clear`) 2nd error
and a few more after that which had troubles during event RegisterEvent dispatch and crash

even after removing every create related mod and leaving only crystal clear and create it crashes - after removing crystal clear mc launched successfully

 took me a while at least to pinpoint the most conflicting mod and hope this 2 logs do help fixing that issue 
 as said alone with create it does work - well, at least it launched and haven't tested it beyond that - but with a couple more, especially in a modpack it does have critical conflicts

**Comment by Cyvack** [comment]
That is strange 
I'll look into why this is happening and see if I can resolve the issue

**Comment by DadoIrie** [comment]
yeah pretty strange - at the first crash log as said it really doesn't matter which mod deactivating (except create itself of course) the crash didn't happened - tried figuring out where the conflict might be which didn't had the effect I wished for

https://mclo.gs/gWTGGrg
[✘] createendertransmission-1.2.3.jar (disabled)
no crash and generated world and works fine even tho ender transmission shouldn't have nothing to do with the crystal clear

https://mclo.gs/rLnQlMY
[✘] yeetusexperimentus-1.0.1-build.2 mc1.19.1.jar (disabled)
wth?!? yeetus just removes the experimental popup before entering the world - still no problems and seems like everything works fine as is should

https://mclo.gs/rOEDTCt
[✔] create_crystal_clear-0.2.a-1.19.2
[✔] create_jetpack-3.1.0
[✔] create-1.19.2-0.5.1.a
[✔] CreateCasing-1.19.2-1.2
[✔] createendertransmission-1.2.3
[✔] kotlinforforge-3.12.0-all
[✔] sleepable_create-1.19.2-0.1.0
[✔] yeetusexperimentus-1.0.1-build.2 mc1.19.1
basically the whole testing instance for create 0.5.1 and it's current addons
crash during launch with this time not the cogwheel but shaft not being present
L193 `Exception caught during firing event: Registry entry not present: create:shaft`

that's what I searched for before, because sometimes its the cogwheel while other times its the shaft
on my modpack I've removed yeetus but still crashes - with same error

**Comment by Cyvack** [comment]
Weird 
I just downloaded all of the mods you listed (except for create casing) and they worked fine for me-
This is very confusing

**Comment by DadoIrie** [comment]
forge 43.2.3 on mc 1.19.2? 
might use also additional some java flags?

**Comment by Cyvack** [comment]
I'm using base java flags, and I'm using that forge and mc version  as well 
Also not in a dev environment, using prism launcher 
All other versions of crystal clear compatible with 0.5.1.a also work for me, so I'm at a lost as to how this could be happening

**Comment by StarsDeath2938** [comment]
Same issue occurs on my side.
Crashed my game with no mod error, i don't know how, but when i remove this from mod folder, my game startup successfully.
Even other mod's registry throw errors, it's always throw create:cogwheel registry error on my side

Forge 1.19.2-43.2.11
Prism Luancher with base java flags

**Comment by DadoIrie** [comment]
Please share log so it can be better looked into it ... just upload it with the prism launcher & share the link here

**Comment by SlimF** [comment]
Don't know if what I'm getting is related but here's my crashlog: 
[crash-2023-05-24_17.40.17-client.txt](https://github.com/NovaStarter/Create-Crystal-Clear/files/11555894/crash-2023-05-24_17.40.17-client.txt)

Game runs fine without Crystal Clear


Edit: I'm blind, didn't see there's newer versions on Modrinth. Game isn't running with latest 0.2.a version as well: [crash-2023-05-24_17.45.32-client.txt](https://github.com/NovaStarter/Create-Crystal-Clear/files/11555945/crash-2023-05-24_17.45.32-client.txt)

**Comment by DadoIrie** [comment]
> Don't know if what I'm getting is related but here's my crashlog: [crash-2023-05-24_17.40.17-client.txt](https://github.com/NovaStarter/Create-Crystal-Clear/files/11555894/crash-2023-05-24_17.40.17-client.txt)
> 
> Game runs fine without Crystal Clear
> 
> Edit: I'm blind, didn't see there's newer versions on Modrinth. Game isn't running with latest 0.2.a version as well: [crash-2023-05-24_17.45.32-client.txt](https://github.com/NovaStarter/Create-Crystal-Clear/files/11555945/crash-2023-05-24_17.45.32-client.txt

`Cannot invoke "net.minecraftforge.registries.IForgeRegistry.getKeys()" because the return value of "java.util.function.Supplier.get()" is null`
unrelated and might not be even related to crystal clear (`ParticleEngine.modifyReceiver$cpj000$visuality$mergeParticleTypes`) - got lucky by removing crystal clear XD

**Comment by SlimF** [comment]
Yep, that was the crashlog with the older version I was using before the edit. 

I now tried with the 0.2.a version of Crystal Clear (and without Visuality just in case) and get this: 
[crash-2023-05-24_19.15.23-client.txt](https://github.com/NovaStarter/Create-Crystal-Clear/files/11556772/crash-2023-05-24_19.15.23-client.txt)

Apologies if it's still unrelated, I won't spam anymore!

**Comment by DadoIrie** [comment]
https://mclo.gs/fYjSTcD#L42 `Create: Crystal Clear (create_crystal_clear) encountered an error during the common_setup event phase §7java.lang.NullPointerException: Registry entry not present: create:cogwheel` <del>which is a `ModLoadingException` tho and not a `NullPointerException` - most likely because of the crash before it was suppressed and had troubles loading - no idea and I hope it does help finding the root of that issue -.-</del>

UPDATE: my bad - the `ModLoadingException` does happen because of the `NullPointerException` as it seems - so crystal clear couldn't load because of the missing reference
seems to me related even tho it's differently thrown

**Comment by Cyvack** [comment]
Been refactoring crystal clear and have been talking to some people to see if the issue can be resolved 

Don't expect it to be soon as I have irl stuff to do

**Comment by Cyvack** [comment]
Issue has been fixed, I'll be pushing a new comit to the github and preparing an upload to modrinth and cf soon
